xbmc/filesystem/BlurayDirectory.cpp

```diff
@@ -75,211 +88,503 @@ std::string CBlurayDirectory::GetBlurayID()
+        videoInfo.codecName = "";
+        break;
+    }
+    switch (video.aspect)
+    {
+      case BLURAY_ASPECT_RATIO_4_3:
+        videoInfo.videoAspectRatio = 4.0f / 3.0f;
+        break;
+      case BLURAY_ASPECT_RATIO_16_9:
+        videoInfo.videoAspectRatio = 16.0f / 9.0f;
+        break;
+      default:
+        videoInfo.videoAspectRatio = 0.0f;
+        break;
+    }
+    videoInfo.stereoMode = ""; // Not stored in BLURAY_TITLE_INFO
+    videoInfo.flags = FLAG_NONE;
+    videoInfo.hdrType = StreamHdrType::HDR_TYPE_NONE; // Not stored in BLURAY_TITLE_INFO
+    videoInfo.fpsRate = 0; // Not in streamdetails
+    videoInfo.fpsScale = 0; // Not in streamdetails
+
+    CVideoInfoTag* info = item->GetVideoInfoTag();
+    info->m_streamDetails.SetStreams(videoInfo, static_cast<int>(title->duration / 90000),
+                                     AudioStreamInfo{}, SubtitleStreamInfo{});
+
+    for (int i = 0; i < title->clips[0].audio_stream_count; ++i)
+    {
+      auto& audio{title->clips[0].audio_streams[i]};
+
+      AudioStreamInfo audioInfo;
+      audioInfo.valid = true;
+      audioInfo.bitrate = 0;
+      audioInfo.channels = 0; // Only basic mono/stereo/multichannel is stored in BLURAY_TITLE_INFO
+
+      switch (audio.coding_type)
+      {
+        case BLURAY_STREAM_TYPE_AUDIO_AC3:
+          audioInfo.codecName = "ac3";
+          break;
+        case BLURAY_STREAM_TYPE_AUDIO_AC3PLUS:
+        case BLURAY_STREAM_TYPE_AUDIO_AC3PLUS_SECONDARY:
+          audioInfo.codecName = "eac3";
+          break;
+        case BLURAY_STREAM_TYPE_AUDIO_LPCM:
+          audioInfo.codecName = "pcm";
+          break;
+        case BLURAY_STREAM_TYPE_AUDIO_DTS:
+          audioInfo.codecName = "dts";
+          break;
+        case BLURAY_STREAM_TYPE_AUDIO_DTSHD:
+        case BLURAY_STREAM_TYPE_AUDIO_DTSHD_SECONDARY:
+          audioInfo.codecName = "dtshd";
+          break;
+        case BLURAY_STREAM_TYPE_AUDIO_DTSHD_MASTER:
+          audioInfo.codecName = "dtshd_ma";
+          break;
+        case BLURAY_STREAM_TYPE_AUDIO_TRUHD:
+          audioInfo.codecName = "truehd";
+          break;
+        default:
+          audioInfo.codecName = "";
+          break;
+      }
+      audioInfo.flags = FLAG_NONE;
+      audioInfo.language = reinterpret_cast<char*>(audio.lang);
+      info->m_streamDetails.AddStream(new CStreamDetailAudio(audioInfo));
+    }
+
+    // Subtitles
+    for (int i = 0; i < title->clips[0].pg_stream_count; ++i)
+    {
+      auto& subtitle{title->clips[0].pg_streams[i]};
+
+      SubtitleStreamInfo subtitleInfo;
+      subtitleInfo.valid = true;
+      subtitleInfo.bitrate = 0;
+      subtitleInfo.codecDesc = "";
+      subtitleInfo.codecName = "";
+      subtitleInfo.isExternal = false;
+      subtitleInfo.name = "";
+      subtitleInfo.flags = FLAG_NONE;
+
+      subtitleInfo.language = reinterpret_cast<char*>(subtitle.lang);
+      info->m_streamDetails.AddStream(new CStreamDetailSubtitle(subtitleInfo));
```

Comment:
```c++
      SubtitleStreamInfo subtitleInfo;
      subtitleInfo.valid = true;
      subtitleInfo.bitrate = 0;
      subtitleInfo.codecDesc = "";
      subtitleInfo.codecName = "";
      subtitleInfo.isExternal = false;
      subtitleInfo.name = "";
      subtitleInfo.flags = FLAG_NONE;

      auto& subtitle{title->clips[0].pg_streams[i]};
      subtitleInfo.language = reinterpret_cast<char*>(subtitle.lang);

      info->m_streamDetails.AddStream(new CStreamDetailSubtitle(subtitleInfo));
```

Reply:
BTW, not related to this PR, looking at all the manual member initialization steps with default values for `*Info`, those structs would really benefit from a default ctor...